pkg/server/api/bundle.go

```diff
@@ -40,3 +42,69 @@ func BundleToProto(b *common.Bundle) (*types.Bundle, error) {
 		JwtAuthorities:  jwtAuthorities,
 	}, nil
 }
+
+func ProtoToBundle(b *types.Bundle) (*common.Bundle, error) {
+	if b == nil {
+		return nil, errors.New("no bundle provided")
+	}
+
+	td, err := spiffeid.TrustDomainFromString(b.TrustDomain)
+	if err != nil && b.TrustDomain != "" {
```

Comment:
In the AppendBundle RPC the `InputMask` is applied before calling `ProtoToBundle(...)`. So, it could be a case where `TrustDomain` flag is not set in the input mask and it is passed as empty to `ProtoToBundle`.

Reply:
maybe we can apply mask after parsing? that makes sense to me, and prevent us to add this kind of code?